apps/aecore/test/aecore_forking_SUITE.erl

```diff
@@ -131,7 +143,89 @@ sync_fork_in_wrong_order(Config) ->
     aec_test_utils:wait_for_it(
       fun() -> rpc:call(N2, aec_chain, top_block, [], 5000) end,
       N1Top),
+    ok = stop_and_check([dev1, dev2], Config).
+
+add_dev3_node(Config) ->
+    %% dev1 and dev2 are in sync
+    T0 = os:timestamp(),
+    aecore_suite_utils:start_node(dev1, Config),
+    aecore_suite_utils:start_node(dev2, Config),
+    [N1, N2, N3] = [aecore_suite_utils:node_name(D) || D <- [dev1, dev2, dev3]],
+    aecore_suite_utils:connect(N1),
+    aecore_suite_utils:connect(N2),
+    aecore_suite_utils:await_sync_complete(T0, [N1, N2]),
+    Top = rpc:call(N1, aec_chain, top_block, [], 5000),
+    Top = rpc:call(N2, aec_chain, top_block, [], 5000),
+    T1 = os:timestamp(),
+    aecore_suite_utils:start_node(dev3, Config),
+    aecore_suite_utils:connect(N3),
+    aecore_suite_utils:await_sync_complete(T1, [N3]),
+    Top = rpc:call(N3, aec_chain, top_block, [], 5000),
+    %% Nodes are all in sync
+    ok = stop_and_check([dev1, dev2, dev3], Config),
+    ok.
+
+dev3_failed_attack(Config) ->
```

Comment:
Looks OK if the fork resistance is in manual mode - in automatic mode it will also work unless a node restarts:
1) dev1, dev2 mine 120 blocks
2) dev3 mines silently 140 more difficult blocks
3) dev2 stops
4) dev3 comes online
5) dev2 starts
Should we do anything in this case?

Reply:
One could imagine adding some config setting that enables fork resistance from the start, _or_ we rely on whitelisting in that case.

Reply:
IMO the whitelist is useful when syncing from genesis - by knowing that we were synced at least once at some point in time we have a gurantee that Top-100 is finalized so we should reject forks starting before Top-100 and reenable resistance once we synced fully.